In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [2]:
p2_col = ['Address','tenure','vehicle','motorcycle']
s1_col = ['Address','days_w','Fasl']
s2_col = ['Address','days_s','Fasl']


# R98

In [3]:
R1398P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1398/R1398P2.csv')
R1398P2.head()

Address  tenure  room  space  construction  material  vehicle  \
0  20001383908       4     3  100.0             3         4        0   
1  20001383911       4     3   70.0             3         7        0   
2  20001383915       4     3  100.0             3         4        1   
3  20001383916       4     4  100.0             3         4        1   
4  20001383920       4     3  100.0             3         4        0   

   motorcycle  cookingfuel  heatingfuel  waterheatingfuel  facilities  
0           1            3            3                 4           7  
1           0            3            3                 4           5  
2           0            3            3                 4           8  
3           0            3            3                 4           7  
4           0            3            3                 4           8

In [4]:
R1398P2 = R1398P2[['Address','tenure','vehicle','motorcycle']]
R1398P2.head()

Address  tenure  vehicle  motorcycle
0  20001383908       4        0           1
1  20001383911       4        0           0
2  20001383915       4        1           0
3  20001383916       4        1           0
4  20001383920       4        0           0

In [5]:
R1398P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1398/R1398P4S01.csv')
R1398P4S01.head()

Address  member  ISCO_w  ISIC_w  hours_w  days_w  income_w_m  \
0  20011395420       1    7111   43900      8.0     2.0     8000000   
1  20001385412       1    9314   41000      7.0     4.0    12800000   
2  20011395423       1    9314   41000      8.0     1.0     2000000   
3  20005389018       1    9314   41000      8.0     2.0     4200000   
4  20013397808       1    9314   41000      8.0     1.0     6500000   

   income_w_y  netincome_w_m  netincome_w_y  Fasl  year  
0   130000000        8000000      130000000     1    98  
1   102400000       12800000      102400000     1    98  
2    45000000        2000000       45000000     1    98  
3    40000000        4200000       40000000     1    98  
4    45000000        6500000       45000000     1    98

In [6]:
R1398P4S01['Fasl'].unique()

array([1, 2, 3, 4])

In [7]:
R1398P4S01 = R1398P4S01[['Address','days_w','Fasl']]
R1398P4S01['Type'] = 'get paid' # حقوق بگیر
R1398P4S01

Address  days_w  Fasl      Type
0      20011395420     2.0     1  get paid
1      20001385412     4.0     1  get paid
2      20011395423     1.0     1  get paid
3      20005389018     2.0     1  get paid
4      20013397808     1.0     1  get paid
...            ...     ...   ...       ...
10815  23005707411     7.0     4  get paid
10816  23003706223     6.0     4  get paid
10817  23003706210     6.0     4  get paid
10818  23003706210     4.0     4  get paid
10819  23003706212     6.0     4  get paid

[10820 rows x 4 columns]

In [8]:
R1398P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1398/R1398P4S02.csv')
R1398P4S02

Address  member  ISCO_s  ISIC_s  status_s  agriculture  hours_s  \
0      20011395420       1    6121    1440         5            1      5.0   
1      20001385423       1    6111    1110         5            1      6.0   
2      20011395407       1    6111    1110         5            1      7.0   
3      20011395407       1    6121    1440         5            1      4.0   
4      20004387823       1    8322   49230         5            2      8.0   
...            ...     ...     ...     ...       ...          ...      ...   
14334  23002703011       1    8322   49220         5            2     12.0   
14335  23002703011       2    5120   47212         5            2      8.0   
14336  23002703814       1    8322   49220         5            2      8.0   
14337  23006708208       1    5120   47510         5            2      8.0   
14338  23001701414       1    5224   47212         5            2     12.0   

       days_s   Total_cost         sale  income_s_y  Fasl  year  
0         7.0   70000000.0  142500000.0    72500000     1    98  
1         5.0   43000000.0  113000000.0    70000000     1    98  
2         5.0   76500000.0  185000000.0   108500000     1    98  
3         7.0  105000000.0  205000000.0   100000000     1    98  
4         2.0   70000000.0  190000000.0   120000000     1    98  
...       ...          ...          ...         ...   ...   ...  
14334     7.0  100000000.0  240000000.0   140000000     4    98  
14335     7.0  340000000.0  470000000.0   130000000     4    98  
14336     5.0   80000000.0  420000000.0   340000000     4    98  
14337     7.0          0.0  184000000.0   184000000     4    98  
14338     7.0  156000000.0  456000000.0   300000000     4    98  

[14339 rows x 13 columns]

In [9]:
R1398P4S02 = R1398P4S02[['Address','days_s','Fasl']]
R1398P4S02['Type'] = 'Self-employed'
R1398P4S02.rename(columns={"days_s":"days_w"}, inplace=True)
R1398P4S02

/var/folders/54/1c_h498x30q9rjn8rdj5py2r0000gn/T/ipykernel_3058/897817094.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  R1398P4S02['Type'] = 'Self-employed'
/var/folders/54/1c_h498x30q9rjn8rdj5py2r0000gn/T/ipykernel_3058/897817094.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  R1398P4S02.rename(columns={"days_s":"days_w"}, inplace=True)


Address  days_w  Fasl           Type
0      20011395420     7.0     1  Self-employed
1      20001385423     5.0     1  Self-employed
2      20011395407     5.0     1  Self-employed
3      20011395407     7.0     1  Self-employed
4      20004387823     2.0     1  Self-employed
...            ...     ...   ...            ...
14334  23002703011     7.0     4  Self-employed
14335  23002703011     7.0     4  Self-employed
14336  23002703814     5.0     4  Self-employed
14337  23006708208     7.0     4  Self-employed
14338  23001701414     7.0     4  Self-employed

[14339 rows x 4 columns]

In [10]:
# Merge S1 & S۲

# R1398P4S02.merge(R1398P4S01 , on='Address')
S1_S2 = pd.concat([R1398P4S01,R1398P4S02], axis='rows')
S1_S2

Address  days_w  Fasl           Type
0      20011395420     2.0     1       get paid
1      20001385412     4.0     1       get paid
2      20011395423     1.0     1       get paid
3      20005389018     2.0     1       get paid
4      20013397808     1.0     1       get paid
...            ...     ...   ...            ...
14334  23002703011     7.0     4  Self-employed
14335  23002703011     7.0     4  Self-employed
14336  23002703814     5.0     4  Self-employed
14337  23006708208     7.0     4  Self-employed
14338  23001701414     7.0     4  Self-employed

[25159 rows x 4 columns]

In [11]:
df = S1_S2.merge(R1398P2 , on='Address')
df

Address  days_w  Fasl           Type  tenure  vehicle  motorcycle
0      20011395420     2.0     1       get paid       4        1           0
1      20011395420     7.0     1  Self-employed       4        1           0
2      20001385412     4.0     1       get paid       4        1           0
3      20011395423     1.0     1       get paid       0        1           1
4      20005389018     2.0     1       get paid       4        0           0
...            ...     ...   ...            ...     ...      ...         ...
25154  23003706220     7.0     4  Self-employed       4        1           0
25155  23002703011     7.0     4  Self-employed       4        0           0
25156  23002703011     7.0     4  Self-employed       4        0           0
25157  23002703814     5.0     4  Self-employed       4        0           0
25158  23001701414     7.0     4  Self-employed       4        0           0

[25159 rows x 7 columns]

In [12]:
R1398P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1398/R1398P3S07.csv')
R1398P3S07.head()

Address   code  purchased    value
0  20011395420  72211          1   800000
1  20011395420  72317          1   700000
2  20001385412  72211          1   700000
3  20001385423  72211          1  1200000
4  20001385423  72317          1   800000

In [13]:
# /Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1398/R1398P3S07.csv
R98 = df.merge(R1398P3S07, on='Address')
R98['Year'] = 1398
R98

Address  days_w  Fasl           Type  tenure  vehicle  motorcycle  \
0      20011395420     2.0     1       get paid       4        1           0   
1      20011395420     2.0     1       get paid       4        1           0   
2      20011395420     7.0     1  Self-employed       4        1           0   
3      20011395420     7.0     1  Self-employed       4        1           0   
4      20001385412     4.0     1       get paid       4        1           0   
...            ...     ...   ...            ...     ...      ...         ...   
56237  23002703011     7.0     4  Self-employed       4        0           0   
56238  23002703011     7.0     4  Self-employed       4        0           0   
56239  23001701414     7.0     4  Self-employed       4        0           0   
56240  23001701414     7.0     4  Self-employed       4        0           0   
56241  23001701414     7.0     4  Self-employed       4        0           0   

        code  purchased    value  Year  
0      72211          1   800000  1398  
1      72317          1   700000  1398  
2      72211          1   800000  1398  
3      72317          1   700000  1398  
4      72211          1   700000  1398  
...      ...        ...      ...   ...  
56237  73216          1  1500000  1398  
56238  73217          1   500000  1398  
56239  73213          1   600000  1398  
56240  73215          1   900000  1398  
56241  73216          1   350000  1398  

[56242 rows x 11 columns]

# U98

In [14]:
U1398P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1398/U1398P3S07.csv')
U1398P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1398/U1398P2.csv')
U1398P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1398/U1398P4S01.csv')
U1398P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1398/U1398P4S02.csv')

In [15]:
U1398P2 = U1398P2[['Address','tenure','vehicle','motorcycle']]
U1398P2.head()

Address  tenure  vehicle  motorcycle
0  10001000108       4        0           0
1  10001000111       5        1           0
2  10001000114       4        0           0
3  10001000117       4        1           0
4  10001000120       4        0           0

In [16]:
U1398P4S01 = U1398P4S01[['Address','days_w','Fasl']]
U1398P4S01['Type'] = 'get paid'
U1398P4S01.head()

Address  days_w  Fasl      Type
0  10004004220     1.0     1  get paid
1  10002002820     6.0     1  get paid
2  10002002823     2.0     1  get paid
3  10002002823     6.0     1  get paid
4  10001000211     5.0     1  get paid

In [17]:
U1398P4S02 = U1398P4S02[['Address','days_s','Fasl']]
U1398P4S02.rename(columns={'days_s':'days_w'}, inplace=True)
U1398P4S02['Type'] = 'Self-employed'
U1398P4S02

Address  days_w  Fasl           Type
0     10002002808     7.0     1  Self-employed
1     10002002815     7.0     1  Self-employed
2     10002002815     4.0     1  Self-employed
3     10006006814     7.0     1  Self-employed
4     10006006812     7.0     1  Self-employed
...           ...     ...   ...            ...
7305  13001375111     7.0     4  Self-employed
7306  13001374515     6.0     4  Self-employed
7307  13001369614     6.0     4  Self-employed
7308  13001376723     5.0     4  Self-employed
7309  13001376708     5.0     4  Self-employed

[7310 rows x 4 columns]

In [18]:
S1_S2_U98 = pd.concat([U1398P4S01,U1398P4S02])

In [19]:
df = S1_S2_U98.merge(U1398P2, on='Address')
df

Address  days_w  Fasl           Type  tenure  vehicle  motorcycle
0      10004004220     1.0     1       get paid       5        0           1
1      10002002820     6.0     1       get paid       4        1           1
2      10002002823     2.0     1       get paid       4        0           1
3      10002002823     6.0     1       get paid       4        0           1
4      10001000211     5.0     1       get paid       4        1           0
...            ...     ...   ...            ...     ...      ...         ...
20056  13002380609     6.0     4  Self-employed       4        0           0
20057  13006382708     6.0     4  Self-employed       4        1           0
20058  13001375111     7.0     4  Self-employed       4        1           0
20059  13001374515     6.0     4  Self-employed       4        1           0
20060  13001376723     5.0     4  Self-employed       4        1           0

[20061 rows x 7 columns]

In [20]:
U98 = U1398P3S07.merge(df, on='Address')
U98['Year'] = 1398
U98

Address   code  purchased    value  days_w  Fasl      Type  tenure  \
0      10004004220  72211          1   100000     1.0     1  get paid       5   
1      10004004220  73217          1   150000     1.0     1  get paid       5   
2      10002002820  72211          1  1300000     6.0     1  get paid       4   
3      10002002823  72211          1   170000     2.0     1  get paid       4   
4      10002002823  72211          1   170000     6.0     1  get paid       4   
...            ...    ...        ...      ...     ...   ...       ...     ...   
50239  13006383411  73212          1   200000     6.0     4  get paid       4   
50240  13006383411  73216          1   400000     6.0     4  get paid       4   
50241  13002381220  73113          1   500000     6.0     4  get paid       4   
50242  13002381220  73215          1   300000     6.0     4  get paid       4   
50243  13002381220  73216          1  1000000     6.0     4  get paid       4   

       vehicle  motorcycle  Year  
0            0           1  1398  
1            0           1  1398  
2            1           1  1398  
3            0           1  1398  
4            0           1  1398  
...        ...         ...   ...  
50239        0           0  1398  
50240        0           0  1398  
50241        0           0  1398  
50242        0           0  1398  
50243        0           0  1398  

[50244 rows x 11 columns]

# R99

In [21]:
R1399P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1399/R1399P2.csv')
R1399P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1399/R1399P3S07.csv')
R1399P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1399/R1399P4S01.csv')
R1399P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1399/R1399P4S02.csv')

In [22]:
R1399P2 = R1399P2[['Address','tenure','vehicle','motorcycle']]

In [23]:
R1399P4S01 = R1399P4S01[['Address','days_w','Fasl']]
R1399P4S01['Type'] = 'get paid'
R1399P4S01.head()

Address  days_w  Fasl      Type
0  20011394629     1.0     1  get paid
1  20011394623     7.0     1  get paid
2  20011394623     2.0     1  get paid
3  20012395826     3.0     1  get paid
4  20006389426     6.0     1  get paid

In [24]:
R1399P4S02 = R1399P4S02[['Address','days_s','Fasl']]
R1399P4S02.rename(columns={'days_s':'days_w'}, inplace=True)
R1399P4S02['Type'] = 'Self-employed'
R1399P4S02

Address  days_w  Fasl           Type
0      20011394626     3.0     1  Self-employed
1      20011394626     7.0     1  Self-employed
2      20011394623     7.0     1  Self-employed
3      20011394614     3.0     1  Self-employed
4      20011394614     7.0     1  Self-employed
...            ...     ...   ...            ...
12941  23006708623     7.0     4  Self-employed
12942  23002704223     3.0     4  Self-employed
12943  23003706213     6.0     4  Self-employed
12944  23002702624     5.0     4  Self-employed
12945  23002702624     7.0     4  Self-employed

[12946 rows x 4 columns]

In [25]:
s1_s2_r99 = pd.concat([R1399P4S01,R1399P4S02])

In [26]:
df = s1_s2_r99.merge(R1399P2, on ='Address')

In [27]:
df

Address  days_w  Fasl           Type  tenure  vehicle  motorcycle
0      20011394629     1.0     1       get paid       4        0           1
1      20011394623     7.0     1       get paid       4        0           0
2      20011394623     2.0     1       get paid       4        0           0
3      20011394623     7.0     1  Self-employed       4        0           0
4      20012395826     3.0     1       get paid       4        0           0
...            ...     ...   ...            ...     ...      ...         ...
23828  23003706215     7.0     4  Self-employed       4        1           0
23829  23003707023     7.0     4  Self-employed       0        0           0
23830  23003706213     6.0     4  Self-employed       4        0           0
23831  23002702624     5.0     4  Self-employed       4        1           1
23832  23002702624     7.0     4  Self-employed       4        1           1

[23833 rows x 7 columns]

In [28]:
R99 = R1399P3S07.merge(df, on = 'Address')
R99['Year'] = 1399
R99.head()

Address   code  purchased   value  days_w  Fasl           Type  tenure  \
0  20011394629  72211          1  300000     1.0     1       get paid       4   
1  20011394629  72214          1  150000     1.0     1       get paid       4   
2  20011394626  72211          1  900000     3.0     1  Self-employed       4   
3  20011394626  72211          1  900000     7.0     1  Self-employed       4   
4  20011394623  73217          1  720000     7.0     1       get paid       4   

   vehicle  motorcycle  Year  
0        0           1  1399  
1        0           1  1399  
2        1           1  1399  
3        1           1  1399  
4        0           0  1399

# U99

In [29]:
U1399P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1399/U1399P2.csv')
U1399P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1399/U1399P3S07.csv')
U1399P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1399/U1399P4S01.csv')
U1399P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1399/U1399P4S02.csv')

In [30]:
U1399P2 = U1399P2[['Address','tenure','vehicle','motorcycle']]

In [31]:
U1399P4S01 = U1399P4S01[['Address','days_w','Fasl']]
U1399P4S01['Type'] = 'get paid'



In [32]:
U1399P4S02 = U1399P4S02[['Address','days_s','Fasl']]
U1399P4S02.rename(columns={'days_s':'days_w'}, inplace=True)
U1399P4S02['Type'] = 'Self-employed'


In [33]:
s1_s2 = pd.concat([U1399P4S01,U1399P4S02])

In [34]:
df = s1_s2.merge(U1399P2, on='Address')

In [35]:
U99 = U1399P3S07.merge(df, on='Address')
U99['Year'] = 1399
U99.head()

Address   code  purchased   value  days_w  Fasl           Type  tenure  \
0  10009008518  72211          1  900000     7.0     1  Self-employed       4   
1  10009008518  72318          1  400000     7.0     1  Self-employed       4   
2  10009008518  73217          1  300000     7.0     1  Self-employed       4   
3  10011009714  72211          1  200000     6.0     1  Self-employed       5   
4  10005004626  73216          1  100000     6.0     1       get paid       4   

   vehicle  motorcycle  Year  
0        0           1  1399  
1        0           1  1399  
2        0           1  1399  
3        0           1  1399  
4        0           0  1399

# R1400

In [36]:
R1400P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1400/R1400P2.csv')
R1400P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1400/R1400P3S07.csv')
R1400P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1400/R1400P4S01.csv')
R1400P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1400/R1400P4S02.csv')

In [37]:
R1400P2 = R1400P2[p2_col]

In [38]:
R1400P4S01 = R1400P4S01[s1_col]
R1400P4S01['Type'] = 'get paid'
R1400P4S01


Address  days_w  Fasl      Type
0      20001384225     7.0     1  get paid
1      20011394620     3.0     1  get paid
2      20001384220     7.0     1  get paid
3      20003387029     2.0     1  get paid
4      20001384223     6.0     1  get paid
...            ...     ...   ...       ...
10726  23006709832     6.0     4  get paid
10727  23002703022     6.0     4  get paid
10728  23002704225     7.0     4  get paid
10729  23002704225     7.0     4  get paid
10730  23003707035     5.0     4  get paid

[10731 rows x 4 columns]

In [39]:
R1400P4S02 = R1400P4S02[['Address','days_s','Fasl']]
R1400P4S02

Address  days_s  Fasl
0      20001384232     7.0     1
1      20011394623     3.0     1
2      20011394623     7.0     1
3      20011394623     7.0     1
4      20003387029     4.0     1
...            ...     ...   ...
12189  23002703032     6.0     4
12190  23003706625     7.0     4
12191  23003706635     6.0     4
12192  23004705026     7.0     4
12193  23004705026     6.0     4

[12194 rows x 3 columns]

In [40]:
R1400P4S02.rename(columns={'days_s':'days_w'}, inplace=True)
R1400P4S02['Type'] = 'Self-employed'


In [41]:
s1_s2_r1400 = pd.concat([R1400P4S01, R1400P4S02])

In [42]:
s1_s2_r1400.head()

Address  days_w  Fasl      Type
0  20001384225     7.0     1  get paid
1  20011394620     3.0     1  get paid
2  20001384220     7.0     1  get paid
3  20003387029     2.0     1  get paid
4  20001384223     6.0     1  get paid

In [43]:
df = s1_s2_r1400.merge(R1400P2, on='Address')

In [44]:
R1400 = R1400P3S07.merge(df, on ='Address')
R1400['Year'] = 1400

In [45]:
R1400.head()

Address   code  purchased    value  days_w  Fasl           Type  \
0  20001384225  73215          1   600000     7.0     1       get paid   
1  20001384232  72211          1  2300000     7.0     1  Self-employed   
2  20001384232  72317          1   650000     7.0     1  Self-employed   
3  20011394620  72211          1   360000     3.0     1       get paid   
4  20011394620  73215          1   800000     3.0     1       get paid   

   tenure  vehicle  motorcycle  Year  
0       4        0           0  1400  
1       4        1           0  1400  
2       4        1           0  1400  
3       4        0           1  1400  
4       4        0           1  1400

# U1400

In [46]:
U1400P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1400/U1400P2.csv')
U1400P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1400/U1400P3S07.csv')
U1400P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1400/U1400P4S01.csv')
U1400P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1400/U1400P4S02.csv')


In [47]:
U1400P2 = U1400P2[p2_col]

In [48]:
U1400P4S01 = U1400P4S01[s1_col]
U1400P4S01['Type'] = 'get paid'


In [49]:
U1400P4S02 = U1400P4S02[s2_col]
U1400P4S02.rename(columns={'days_s':'days_w'}, inplace=True)
U1400P4S02['Type'] = 'Self-employed'


In [50]:
s1_s2_u1400 = pd.concat([U1400P4S01, U1400P4S02])

In [51]:
df = s1_s2_u1400.merge(U1400P2, on='Address')

In [52]:
U1400 = U1400P3S07.merge(df, on ='Address')
U1400['Year'] = 1400

In [53]:
U1400

Address   code  purchased    value  days_w  Fasl           Type  \
0      10003003229  72211          1  1500000     6.0     1       get paid   
1      10003003229  72211          1  1500000     4.0     1  Self-employed   
2      10003003229  72211          1  1500000     7.0     1  Self-employed   
3      10003003229  73113          1   200000     6.0     1       get paid   
4      10003003229  73113          1   200000     4.0     1  Self-employed   
...            ...    ...        ...      ...     ...   ...            ...   
42665  13001371823  73216          1  1000000     4.0     4  Self-employed   
42666  13001371823  73217          1   600000     6.0     4       get paid   
42667  13001371823  73217          1   600000     4.0     4  Self-employed   
42668  13001371823  73224          1  1500000     6.0     4       get paid   
42669  13001371823  73224          1  1500000     4.0     4  Self-employed   

       tenure  vehicle  motorcycle  Year  
0           4        1           0  1400  
1           4        1           0  1400  
2           4        1           0  1400  
3           4        1           0  1400  
4           4        1           0  1400  
...       ...      ...         ...   ...  
42665       1        1           0  1400  
42666       1        1           0  1400  
42667       1        1           0  1400  
42668       1        1           0  1400  
42669       1        1           0  1400  

[42670 rows x 11 columns]

# R1401

In [54]:
R1401P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1401/R1401P2.csv')
R1401P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1401/R1401P3S07.csv')
R1401P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1401/R1401P4S01.csv')
R1401P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/R1401/R1401P4S02.csv')

In [55]:
R1401P2 = R1401P2[p2_col]

In [56]:
R1401P4S01 = R1401P4S01[s1_col]
R1401P4S01['Type'] = 'get paid'

In [57]:
R1401P4S02 = R1401P4S02[s2_col]
R1401P4S02['Type'] = 'Self-employed'
R1401P4S02.rename(columns={'days_s':'days_w'}, inplace=True)

In [58]:
s1_s2_r1401 = pd.concat([R1401P4S01, R1401P4S02])

In [59]:
df = s1_s2_r1401.merge(R1401P2, on='Address')

In [60]:
R1401 = R1401P3S07.merge(df, on='Address')
R1401['Year'] = 1401

In [61]:
R1401.head()

Address   code  purchased    value  days_w  Fasl           Type  \
0  20001384229  73212          1  1000000       3     1       get paid   
1  20001384229  73212          1  1000000       6     1       get paid   
2  20001384232  72211          1  1200000       7     1  Self-employed   
3  20001384232  72317          1  2500000       7     1  Self-employed   
4  20001384232  72318          1  1000000       7     1  Self-employed   

   tenure  vehicle  motorcycle  Year  
0       5        0           0  1401  
1       5        0           0  1401  
2       4        1           0  1401  
3       4        1           0  1401  
4       4        1           0  1401

# U1401

In [62]:
U1401P2 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1401/U1401P2.csv')
U1401P3S07 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1401/U1401P3S07.csv')
U1401P4S01 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1401/U1401P4S01.csv')
U1401P4S02 = pd.read_csv('/Users/profosor/Downloads/Quera_Project_1/Iran_family_income_analysis-_g6/U1401/U1401P4S02.csv')

In [63]:
U1401P2 = U1401P2[p2_col]

In [64]:
U1401P4S01 = U1401P4S01[s1_col]
U1401P4S01['Type'] = 'get paid'

In [65]:
U1401P4S02 = U1401P4S02[s2_col]
U1401P4S02['Type'] = 'Self-employed'
U1401P4S02.rename(columns={'days_s':'days_w'}, inplace=True)

In [66]:
s1_s2_u1401 = pd.concat([U1401P4S01, U1401P4S02])

In [67]:
df = s1_s2_u1401.merge(U1401P2, on='Address')

In [68]:
U1401 = U1401P3S07.merge(df, on='Address')
U1401['Year'] = 1401

In [69]:
U1401.head()

Address   code  purchased    value  days_w  Fasl      Type  tenure  \
0  10001000126  72211          1  1200000       6     4  get paid       4   
1  10001000126  72211          1  1200000       6     4  get paid       4   
2  10001000126  73212          1   200000       6     4  get paid       4   
3  10001000126  73212          1   200000       6     4  get paid       4   
4  10001000126  73217          1   600000       6     4  get paid       4   

   vehicle  motorcycle  Year  
0        1           0  1401  
1        1           0  1401  
2        1           0  1401  
3        1           0  1401  
4        1           0  1401

# combine R98/R99/R1400/R1401

In [70]:
R_df = pd.concat([R98, R99, R1400, R1401])
R_df

Address  days_w  Fasl           Type  tenure  vehicle  motorcycle  \
0      20011395420     2.0     1       get paid       4        1           0   
1      20011395420     2.0     1       get paid       4        1           0   
2      20011395420     7.0     1  Self-employed       4        1           0   
3      20011395420     7.0     1  Self-employed       4        1           0   
4      20001385412     4.0     1       get paid       4        1           0   
...            ...     ...   ...            ...     ...      ...         ...   
44906  23006709741     7.0     3       get paid       4        1           0   
44907  23006709741     7.0     3       get paid       4        1           0   
44908  23006709741     7.0     3       get paid       4        1           0   
44909  23006709741     7.0     3       get paid       4        1           0   
44910  23006709741     7.0     3       get paid       4        1           0   

        code  purchased    value  Year  
0      72211          1   800000  1398  
1      72317          1   700000  1398  
2      72211          1   800000  1398  
3      72317          1   700000  1398  
4      72211          1   700000  1398  
...      ...        ...      ...   ...  
44906  72211          1  4000000  1401  
44907  72214          1  8000000  1401  
44908  73216          1  1000000  1401  
44909  73217          1  1500000  1401  
44910  73224          1   500000  1401  

[197814 rows x 11 columns]

In [71]:
R_df.describe()

Address         days_w           Fasl         tenure  \
count  1.978140e+05  197814.000000  197814.000000  197814.000000   
mean   2.143900e+10       5.300684       2.490147       3.793291   
std    9.208905e+08       1.629535       1.108448       0.992609   
min    2.000138e+10       0.000000       1.000000       0.000000   
25%    2.060245e+10       4.000000       1.000000       4.000000   
50%    2.140255e+10       6.000000       3.000000       4.000000   
75%    2.230663e+10       7.000000       3.000000       4.000000   
max    2.300671e+10       7.000000       4.000000       6.000000   

             vehicle     motorcycle           code      purchased  \
count  197814.000000  197814.000000  197814.000000  197814.000000   
mean        0.481114       0.436713   72704.927811       1.022931   
std         0.499644       0.495980     491.851723       0.349218   
min         0.000000       0.000000   72211.000000       1.000000   
25%         0.000000       0.000000   72211.000000       1.000000   
50%         0.000000       0.000000   72318.000000       1.000000   
75%         1.000000       1.000000   73215.000000       1.000000   
max         1.000000       1.000000   73511.000000       8.000000   

              value           Year  
count  1.978140e+05  197814.000000  
mean   1.008635e+06    1399.413378  
std    1.762196e+06       1.124816  
min    5.000000e+03    1398.000000  
25%    3.000000e+05    1398.000000  
50%    5.000000e+05    1399.000000  
75%    1.200000e+06    1400.000000  
max    1.500000e+08    1401.000000

In [72]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder

In [73]:
# Encoding categorical variables
# label_encoder = LabelEncoder()
# R_df['Type'] = label_encoder.fit_transform(R_df['Type'])

In [74]:
R_df.corr(numeric_only=True)

Address    days_w      Fasl    tenure   vehicle  motorcycle  \
Address     1.000000  0.056098 -0.011538  0.001118 -0.029244    0.062423   
days_w      0.056098  1.000000 -0.021648  0.002639  0.060035    0.006671   
Fasl       -0.011538 -0.021648  1.000000  0.009319  0.024793   -0.059134   
tenure      0.001118  0.002639  0.009319  1.000000  0.058893    0.022089   
vehicle    -0.029244  0.060035  0.024793  0.058893  1.000000   -0.157269   
motorcycle  0.062423  0.006671 -0.059134  0.022089 -0.157269    1.000000   
code        0.025721 -0.034553  0.003396 -0.032032 -0.500529   -0.172187   
purchased   0.010897  0.004309  0.006304  0.001147 -0.032024   -0.012461   
value      -0.042934  0.020582  0.055594  0.014997  0.184661   -0.040777   
Year       -0.005325 -0.012065  0.027866  0.003274  0.052462   -0.038819   

                code  purchased     value      Year  
Address     0.025721   0.010897 -0.042934 -0.005325  
days_w     -0.034553   0.004309  0.020582 -0.012065  
Fasl        0.003396   0.006304  0.055594  0.027866  
tenure     -0.032032   0.001147  0.014997  0.003274  
vehicle    -0.500529  -0.032024  0.184661  0.052462  
motorcycle -0.172187  -0.012461 -0.040777 -0.038819  
code        1.000000   0.060212 -0.131333 -0.050922  
purchased   0.060212   1.000000  0.010942  0.001041  
value      -0.131333   0.010942  1.000000  0.214469  
Year       -0.050922   0.001041  0.214469  1.000000

### Interpretation and Insights from the Correlation Matrix

correlation matrix with focus on 'value':

| Feature    | Correlation with 'value' |
|------------|--------------------------|
| Address    | -0.042934                |
| days_w     | 0.020582                 |
| Fasl       | 0.055594                 |
| Type       | 0.012211                 |
| tenure     | 0.014997                 |
| vehicle    | 0.184661                 |
| motorcycle | -0.040777                |
| code       | -0.131333                |
| purchased  | 0.010942                 |
| Year       | 0.214469                 |

### Insights:

1. **Year (0.214469)**: The 'Year' feature has the highest positive correlation with 'value'. This indicates that as the year increases, the 'value' tends to increase as well. This might be due to inflation or appreciation of assets over time.

2. **vehicle (0.184661)**: The number of vehicles is positively correlated with 'value'. More vehicles might indicate higher wealth, leading to higher 'value'.

3. **code (-0.131333)**: There is a moderate negative correlation between 'code' and 'value'. This could imply that certain codes are associated with lower values, which might be due to specific regions or categories represented by the codes.

4. **Fasl (0.055594)**: This feature shows a weak positive correlation with 'value'. 'Fasl' might represent some seasonal or cyclical factors that slightly influence the value.

5. **days_w (0.020582)**, **tenure (0.014997)**, **Type (0.012211)**, and **purchased (0.010942)**: These features have very weak correlations with 'value'. They do not have a strong linear relationship with the target variable.

6. **Address (-0.042934)** and **motorcycle (-0.040777)**: These features have a weak negative correlation with 'value'. The 'Address' feature might be too unique or granular to show a meaningful correlation, and owning a motorcycle seems to have a slight negative impact on 'value'.

In [75]:
# R_df = R_df[['days_w','Fasl','Type','vehicle','code','purchased','value','Year']]

In [76]:
import seaborn as sns
import matplotlib.pyplot as plt

# # Correlation matrix
# correlation_matrix = R_df.corr(numeric_only=True)
# sns.heatmap(correlation_matrix, annot=True)
# plt.show()

# Pairplot
# sns.pairplot(R_df)
# plt.show()


### Feature Selection

In [77]:
# X = R_df.drop(['value','Address'], axis=1)
# y = R_df['value']

In [78]:
# from sklearn.ensemble import RandomForestRegressor
# from sklearn.feature_selection import RFE



# # Use a random forest to rank the importance of features
# model = RandomForestRegressor()
# rfe = RFE(model, n_features_to_select=5)
# fit = rfe.fit(X, y)

# print("Selected Features: %s" % X.columns[fit.support_])


###  Random Forest with scaling

In [79]:
# import pandas as pd
# from sklearn.model_selection import train_test_split, GridSearchCV
# from sklearn.metrics import mean_squared_error, r2_score
# from sklearn.ensemble import RandomForestRegressor
# from sklearn.preprocessing import StandardScaler
# from sklearn.pipeline import Pipeline

# # Split the data into training and testing sets
# X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# # Define the scaling and model pipeline
# pipeline = Pipeline([
#     ('scaler', StandardScaler()), 
#     ('model', RandomForestRegressor())  
# ])

# # Define the parameter grid for grid search
# param_grid = {
#     'model__n_estimators': [50, 100, 150, 200],
#     'model__max_depth': [10, 20, 30],
#     'model__min_samples_split': [2, 5, 4,6]
# }

# # Perform grid search
# grid_search = GridSearchCV(estimator=pipeline, param_grid=param_grid, cv=3, verbose=2 , n_jobs=-1)
# grid_search.fit(X_train, y_train)

# # Best parameters
# print("Best parameters: ", grid_search.best_params_)

# # Predict on test data
# y_pred = grid_search.predict(X_test)

# # Evaluate the model
# mse = mean_squared_error(y_test, y_pred)
# rmse = mse ** 0.5
# r2 = r2_score(y_test, y_pred)

# print(f'MSE: {mse}, RMSE: {rmse}, R-squared: {r2}')

### XGBoost Regressor

In [80]:
# from xgboost import XGBRegressor

# # Define the scaling and model pipeline
# pipeline = Pipeline([
#     ('scaler', StandardScaler()),
#     ('model', XGBRegressor())
# ])

# # Define the parameter grid for grid search
# param_grid = {
#     'model__n_estimators': [100, 200, 300],
#     'model__learning_rate': [0.01, 0.1, 0.2],
#     'model__max_depth': [3, 4, 5]
# }

# # Perform grid search
# grid_search = GridSearchCV(estimator=pipeline, param_grid=param_grid, cv=3, verbose=1)
# grid_search.fit(X_train, y_train)

# # Best parameters
# print("Best parameters: ", grid_search.best_params_)

# # Predict on test data
# y_pred = grid_search.predict(X_test)

# # Evaluate the model
# mse = mean_squared_error(y_test, y_pred)
# rmse = mse ** 0.5
# r2 = r2_score(y_test, y_pred)

# print(f'MSE: {mse}, RMSE: {rmse}, R-squared: {r2}')


<!-- from xgboost import XGBRegressor
from sklearn.model_selection import GridSearchCV
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import mean_squared_error, r2_score

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Define the scaling and model pipeline
pipeline = Pipeline([
    ('scaler', StandardScaler()),
    ('model', XGBRegressor())
])

# Define the expanded parameter grid for grid search
param_grid = {
    'model__n_estimators': [100, 200, 300, 400, 500],
    'model__learning_rate': [0.01, 0.05, 0.1, 0.2, 0.3],
    'model__max_depth': [3, 4, 5, 6, 7],
    'model__subsample': [0.6, 0.7, 0.8, 0.9, 1.0],
    'model__colsample_bytree': [0.6, 0.7, 0.8, 0.9, 1.0],
    'model__min_child_weight': [1, 2, 3, 4, 5]
}

# Perform grid search
grid_search = GridSearchCV(estimator=pipeline, param_grid=param_grid, cv=3, n_jobs=-1, verbose=2)
grid_search.fit(X_train, y_train)

# Best parameters
print("Best parameters: ", grid_search.best_params_)

# Predict on test data
y_pred = grid_search.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
rmse = mse ** 0.5
r2 = r2_score(y_test, y_pred)

print(f'MSE: {mse}, RMSE: {rmse}, R-squared: {r2}')
 -->

# Combine U98/U99/U1400/U1401

In [81]:
U_df = pd.concat([U98, U99, U1400, U1401])

# Combine R & U

In [82]:
R_df['R&U'] = 'R'

In [83]:
U_df['R&U'] = 'U'

In [84]:
U_df.head()

Address   code  purchased    value  days_w  Fasl      Type  tenure  \
0  10004004220  72211          1   100000     1.0     1  get paid       5   
1  10004004220  73217          1   150000     1.0     1  get paid       5   
2  10002002820  72211          1  1300000     6.0     1  get paid       4   
3  10002002823  72211          1   170000     2.0     1  get paid       4   
4  10002002823  72211          1   170000     6.0     1  get paid       4   

   vehicle  motorcycle  Year R&U  
0        0           1  1398   U  
1        0           1  1398   U  
2        1           1  1398   U  
3        0           1  1398   U  
4        0           1  1398   U

In [85]:
R_df.head()

Address  days_w  Fasl           Type  tenure  vehicle  motorcycle  \
0  20011395420     2.0     1       get paid       4        1           0   
1  20011395420     2.0     1       get paid       4        1           0   
2  20011395420     7.0     1  Self-employed       4        1           0   
3  20011395420     7.0     1  Self-employed       4        1           0   
4  20001385412     4.0     1       get paid       4        1           0   

    code  purchased   value  Year R&U  
0  72211          1  800000  1398   R  
1  72317          1  700000  1398   R  
2  72211          1  800000  1398   R  
3  72317          1  700000  1398   R  
4  72211          1  700000  1398   R

In [86]:
Main_df = pd.concat([R_df,U_df]).reset_index()

In [87]:
Main_df.head()

index      Address  days_w  Fasl           Type  tenure  vehicle  \
0      0  20011395420     2.0     1       get paid       4        1   
1      1  20011395420     2.0     1       get paid       4        1   
2      2  20011395420     7.0     1  Self-employed       4        1   
3      3  20011395420     7.0     1  Self-employed       4        1   
4      4  20001385412     4.0     1       get paid       4        1   

   motorcycle   code  purchased   value  Year R&U  
0           0  72211          1  800000  1398   R  
1           0  72317          1  700000  1398   R  
2           0  72211          1  800000  1398   R  
3           0  72317          1  700000  1398   R  
4           0  72211          1  700000  1398   R

In [88]:
# Encoding categorical variables
label_encoder = LabelEncoder()
Main_df['Type'] = label_encoder.fit_transform(Main_df['Type'])
Main_df['R&U'] = label_encoder.fit_transform(Main_df['R&U'])

In [89]:
Main_df.head()

index      Address  days_w  Fasl  Type  tenure  vehicle  motorcycle   code  \
0      0  20011395420     2.0     1     1       4        1           0  72211   
1      1  20011395420     2.0     1     1       4        1           0  72317   
2      2  20011395420     7.0     1     0       4        1           0  72211   
3      3  20011395420     7.0     1     0       4        1           0  72317   
4      4  20001385412     4.0     1     1       4        1           0  72211   

   purchased   value  Year  R&U  
0          1  800000  1398    0  
1          1  700000  1398    0  
2          1  800000  1398    0  
3          1  700000  1398    0  
4          1  700000  1398    0

In [90]:
X = Main_df.drop(['value','Address'], axis=1)
y = Main_df['value']

# Randoem Forest (R&U)

In [94]:
import pandas as pd
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.metrics import mean_squared_error, r2_score
from sklearn.ensemble import RandomForestRegressor
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Define the scaling and model pipeline
pipeline = Pipeline([
    ('scaler', StandardScaler()), 
    ('model', RandomForestRegressor())  
])

# Define the parameter grid for grid search
param_grid = {
    'model__n_estimators': [50, 100, 150, 200],
    'model__max_depth': [10, 20, 30],
    'model__min_samples_split': [2, 5 ,10]
}

# Perform grid search
grid_search = GridSearchCV(estimator=pipeline, param_grid=param_grid, cv=5, verbose=1 , n_jobs=-1)
grid_search.fit(X_train, y_train)

# Best parameters
print("Best parameters: ", grid_search.best_params_)

# Predict on test data
y_pred = grid_search.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
rmse = mse ** 0.5
r2 = r2_score(y_test, y_pred)

print(f'MSE: {mse}, RMSE: {rmse}, R-squared: {r2}')

Fitting 5 folds for each of 36 candidates, totalling 180 fits
[CV] END model__max_depth=10, model__min_samples_split=2, model__n_estimators=50; total time=  27.3s
[CV] END model__max_depth=10, model__min_samples_split=2, model__n_estimators=100; total time=  55.8s
[CV] END model__max_depth=10, model__min_samples_split=2, model__n_estimators=200; total time= 1.8min
[CV] END model__max_depth=10, model__min_samples_split=5, model__n_estimators=100; total time=  53.2s
[CV] END model__max_depth=10, model__min_samples_split=5, model__n_estimators=200; total time= 1.8min
[CV] END model__max_depth=10, model__min_samples_split=10, model__n_estimators=100; total time=  54.1s
[CV] END model__max_depth=10, model__min_samples_split=10, model__n_estimators=150; total time= 1.3min
[CV] END model__max_depth=20, model__min_samples_split=2, model__n_estimators=50; total time=  40.8s
[CV] END model__max_depth=20, model__min_samples_split=2, model__n_estimators=100; total time= 1.3min


/opt/anaconda3/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:700: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.
  warnings.warn(


[CV] END model__max_depth=10, model__min_samples_split=2, model__n_estimators=100; total time=  54.9s
[CV] END model__max_depth=10, model__min_samples_split=2, model__n_estimators=200; total time= 1.8min
[CV] END model__max_depth=10, model__min_samples_split=5, model__n_estimators=100; total time=  53.3s
[CV] END model__max_depth=10, model__min_samples_split=5, model__n_estimators=150; total time= 1.4min
[CV] END model__max_depth=10, model__min_samples_split=10, model__n_estimators=50; total time=  27.0s
[CV] END model__max_depth=10, model__min_samples_split=10, model__n_estimators=100; total time=  54.4s
[CV] END model__max_depth=10, model__min_samples_split=10, model__n_estimators=150; total time= 1.4min
[CV] END model__max_depth=20, model__min_samples_split=2, model__n_estimators=50; total time=  41.2s
[CV] END model__max_depth=20, model__min_samples_split=2, model__n_estimators=50; total time=  40.2s
[CV] END model__max_depth=20, model__min_samples_split=2, model__n_estimators=150;

In [93]:
from xgboost import XGBRegressor

# Define the scaling and model pipeline
pipeline = Pipeline([
    ('scaler', StandardScaler()),
    ('model', XGBRegressor())
])

# Define the parameter grid for grid search
param_grid = {
    'model__n_estimators': [100, 200, 300],
    'model__learning_rate': [0.01, 0.1, 0.2, 0.4],
    'model__max_depth': [3, 4, 5]
}

# Perform grid search
grid_search = GridSearchCV(estimator=pipeline, param_grid=param_grid, cv=5, verbose=1)
grid_search.fit(X_train, y_train)

# Best parameters
print("Best parameters: ", grid_search.best_params_)

# Predict on test data
y_pred = grid_search.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
rmse = mse ** 0.5
r2 = r2_score(y_test, y_pred)

print(f'MSE: {mse}, RMSE: {rmse}, R-squared: {r2}')


Fitting 5 folds for each of 36 candidates, totalling 180 fits
Best parameters:  {'model__learning_rate': 0.4, 'model__max_depth': 5, 'model__n_estimators': 300}
MSE: 2004783251678.5886, RMSE: 1415903.6872890007, R-squared: 0.3999508856747226
